src/nspawn/nspawn.c

```diff
@@ -2594,9 +2594,15 @@ static int inner_child(
                 return -ESRCH;
         }
 
-        r = mount_systemd_cgroup_writable("", arg_unified_cgroup_hierarchy);
-        if (r < 0)
-                return r;
+	if (cg_ns_supported()) {
+		r = unshare(CLONE_NEWCGROUP);
+		if (r < 0)
```

Comment:
Thanks for testing on unified @evverx. Starting unified with `systemd-nspawn` didn't work for me with `v228` independent of the patch. So maybe I need to test that from master again.

Your second point I'm not entirely clear what you're getting at. In the case of cgroup namespaces the container will be able to mount a cgroup filesystem by itself just as on normal system bootup. So we don't need to bind-mount, I think. If you're getting at the point about some subsystems missing in the container. This is explained by how cgroup v1 and v2 interact I think: As you have mounted `cgroup2` on the host you likely have mounted the available subsystems `memory`, `pid` etc. into the `v2` hierarchy which means that they are not mounted into the `v1` hierarchy. This is why they do not appear in the container which checks the available controllers in the v1 hierarchy.

@poettering would you prefer a different approach?